File: services/profile-api/Models/Friendship.cs
using profile_api.Models.Enums;

namespace profile_api.Models;

public class Friendship
{
    public Guid Id { get; set; }
    public Guid RequesterProfileId { get; set; }
    public Guid ReceiverProfileId { get; set; }
    public DateTime CreatedDate { get; set; }
    
    public Profile? RequesterProfile { get; set; }
    public Profile? ReceiverProfile { get; set; }
    public FriendshipState FriendshipState { get; set; }
}

File: clients/web/garfy-web/src/contracts/profile/friendship.ts
import Profile from "./profile";

export interface Friendship {
  id: string;
  requesterProfileId: string;
  receiverProfileId: string;
  requesterProfile: Profile;
  receiverProfile: Profile;
  createdDate: Date;
  friendshipState: FriendshipState;
}

export enum FriendshipState {
  Pending,
  Accepted,
  Rejected,
}
